File: MercuryClient/Globalization/Territory.cs
/* 
 * Copyright (C) 2013 Alex Bikfalvi
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or (at
 * your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 */

using System;

namespace Mercury.Globalization
{
	/// <summary>
	/// A class representing a territory.
	/// </summary>
	[Serializable]
	public class Territory
	{
		/// <summary>
		/// Creates a new territory instance.
		/// </summary>
		/// <param name="type">The territory type.</param>
		/// <param name="name">The laguage name.</param>
		public Territory(string type, string name)
		{
			this.Type = type.ToLowerInvariant();
			this.Name = name;
		}

		// Public properties.

		/// <summary>
		/// Gets the territory type.
		/// </summary>
		public string Type { get; private set; }
		/// <summary>
		/// Gets the territory name.
		/// </summary>
		public string Name { get; private set; }

		// Public methods.

		/// <summary>
		/// Gets the territory name.
		/// </summary>
		/// <returns>The name.</returns>
		public override string ToString()
		{
			return this.Name;
		}

		/// <summary>
		/// Compares with an object for equality.
		/// </summary>
		/// <param name="obj">The object to compare.</param>
		/// <returns><b>True</b> if the objects are the equal, <b>false</b> otherwise.</returns>
		public override bool Equals(object obj)
		{
			if (null == obj) return false;
			Territory territory = obj as Territory;
			if (null == territory) return false;
			return this.Type == territory.Type;
		}

		/// <summary>
		/// Returns the hash code of the current object.
		/// </summary>
		/// <returns>The hash code.</returns>
		public override int GetHashCode()
		{
			return this.Type.GetHashCode();
		}

		/// <summary>
		/// Compares two territory objects for equality.
		/// </summary>
		/// <param name="left">The left territory.</param>
		/// <param name="right">The right territory.</param>
		/// <returns><b>True</b> if the territories are the equal, <b>false</b> otherwise.</returns>
		public static bool operator ==(Territory left, Territory right)
		{
			if (object.ReferenceEquals(left, right)) return true;
			if (((object)left == null) || ((object)right == null)) return false;
			return left.Type == right.Type;
		}

		/// <summary>
		/// Compares two territory objects for inequality.
		/// </summary>
		/// <param name="left">The left territory.</param>
		/// <param name="right">The right territory.</param>
		/// <returns><b>True</b> if the territories are the different, <b>false</b> otherwise.</returns>
		public static bool operator !=(Territory left, Territory right)
		{
			return !(left == right);
		}
	}
}

File: Eclipse/MercuryClient (old)/src/com/bikfalvi/java/globalization/Territory.java
/** 
 * Copyright (C) 2013 Alex Bikfalvi
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or (at
 * your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 */

package com.bikfalvi.java.globalization;

import java.util.Locale;

/**
 * A class representing a territory.
 * @author Alex Bikfalvi
 *
 */
public class Territory {
	private String type;
	private String name;
	
	/**
	 * Creates a new territory instance.
	 * @param type The territory type.
	 * @param name The territory name.
	 */
	public Territory(String type, String name) {
		this.type = type.toLowerCase(Locale.US);
		this.name = name;
	}
	
	/**
	 * Gets the territory type.
	 * @return The type.
	 */
	public String getType() {
		return type;
	}

	/**
	 * Sets the territory type.
	 * @param type The type.
	 */
	public void setType(String type) {
		this.type = type;
	}

	/**
	 * Gets the territory name.
	 * @return The name.
	 */
	public String getName() {
		return name;
	}

	/**
	 * Sets the territory name.
	 * @param name The name.
	 */
	public void setName(String name) {
		this.name = name;
	}
	
	@Override
	public boolean equals(Object object) {
		if (null == object) return false;
		if (object instanceof Territory) {
			Territory territory = (Territory)object;
			return this.type.equals(territory.type);
		}
		else if (object instanceof String) {
			String territory = (String) object;
			return this.type.equals(territory);
		}
		else return false;
	}
	
	@Override
	public int hashCode() {
		return this.type.hashCode();
	}

	@Override
	public String toString() {
		return this.name;
	}
}
